### PR TITLE
treewide (darwin): fix or enable darwin build for many packages (2)

### PR DESCRIPTION
**Pull request by siraben**
###### Motivation for this change
See #114976 for details.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 115051` at 87a0168f run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/6d79a3651c98995f5196f0a51a1bd8c9">cconv</a></li></li>
    <li><a href="https://gist.github.com/225eb37408ab09481e4db1a18a3473ca">clips</a></li></li>
    <li><a href="https://gist.github.com/c5f8e6ee34fd56379b2f946d997ca0cc">dadadodo</a></li></li>
    <li><a href="https://gist.github.com/56e42b0129cf12916a333c81162d6e2c">dos2unix</a></li></li>
    <li><a href="https://gist.github.com/ee160189a662c343637f5473b01e44bb">vimv</a></li></li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/license-missing.md">license-missing</a>

Package is missing a license

Near pkgs/tools/text/dadadodo/default.nix:21:3:
```
   |
21 |   meta = with lib; {
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/tools/text/dadadodo/default.nix:14:3:
```
   |
14 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 115051` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>18 packages built:</summary>
  <ul>
    <li>aespipe</li>
    <li>agedu</li>
    <li>anthy</li>
    <li>boolstuff</li>
    <li>cconv</li>
    <li>clips</li>
    <li>cpptest</li>
    <li>cpputest</li>
    <li>dadadodo</li>
    <li>direvent</li>
    <li>dos2unix</li>
    <li>fanficfare</li>
    <li>libdsk</li>
    <li>loadwatch</li>
    <li>mednafen-server</li>
    <li>pcapfix</li>
    <li>podiff</li>
    <li>vimv</li>
  </ul>
</details>

File: pkgs/tools/misc/vimv/default.nix
Comment (SuperSandro2000) [inline pkgs/tools/misc/vimv/default.nix:3]: When should stdenvNoCC be used?

File: pkgs/tools/security/hashdeep/default.nix
Comment (SuperSandro2000) [inline pkgs/tools/security/hashdeep/default.nix:23]:
```suggestion
    broken = stdenv.isDarwin;
    platforms = platforms.all;
```

File: pkgs/tools/text/cconv/default.nix
Comment (SuperSandro2000) [inline pkgs/tools/text/cconv/default.nix:13]: Is this only required for darwin?

File: pkgs/tools/text/fanficfare/default.nix
Comment (SuperSandro2000) [inline pkgs/tools/text/fanficfare/default.nix:27]:
```suggestion
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/vimv/default.nix
Comment (siraben) [reply pkgs/tools/misc/vimv/default.nix:3]: I checked the source and there is no invocation of a compiler, it's just a shell script. See https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/misc/neofetch/default.nix for instance.

File: pkgs/tools/text/fanficfare/default.nix
Comment (siraben) [reply pkgs/tools/text/fanficfare/default.nix:27]: @jonringer should version be inherited?

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 115051` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package marked as broken and skipped:
</summary>
<ul>
<li>hashdeep</li>
</ul>
</details>
<details>
<summary>
18 packages built:
</summary>
<ul>
<li>aespipe</li>
<li>agedu</li>
<li>anthy</li>
<li>boolstuff</li>
<li>cconv</li>
<li>clips</li>
<li>cpptest</li>
<li>cpputest</li>
<li>dadadodo</li>
<li>direvent</li>
<li>dos2unix</li>
<li>fanficfare</li>
<li>libdsk</li>
<li>loadwatch</li>
<li>mednafen-server</li>
<li>pcapfix</li>
<li>podiff</li>
<li>vimv</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
aespipe:
<details>


warning: unclear-gpl
`gpl2` is a deprecated license, check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/tools/security/aespipe/default.nix:25:5:

```
   |
25 |     license = licenses.gpl2;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
cconv:
<details>

warning: EvalError
Cannot evaluate attribute ‘cconv’ in ‘/nix/store/l6cbj3mm18s3a8wbnvx2aafbaig3z854-nixpkgs’.
</details>
cpptest:
<details>


warning: unclear-gpl
`lgpl3` is a deprecated license, check if project uses `lgpl3Plus` or `lgpl3Only` and change `meta.license` accordingly.

Near pkgs/development/libraries/cpptest/default.nix:15:5:

```
   |
15 |     license = lib.licenses.lgpl3;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
dadadodo:
<details>


warning: license-missing
Package is missing a license

Near pkgs/tools/text/dadadodo/default.nix:21:3:

```
   |
21 |   meta = with lib; {
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/license-missing.md
</details>
libdsk:
<details>


warning: maintainers-missing
Package does not have a maintainer. Consider adding yourself?

Near pkgs/misc/emulators/libdsk/default.nix:16:5:

```
   |
16 |     maintainers = [ ];
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
</details>
loadwatch:
<details>


warning: unclear-gpl
`gpl2` is a deprecated license, check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/tools/system/loadwatch/default.nix:16:5:

```
   |
16 |     license = licenses.gpl2;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
mednafen-server:
<details>


warning: unclear-gpl
`gpl2` is a deprecated license, check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/misc/emulators/mednafen/server.nix:17:5:

```
   |
17 |     license = licenses.gpl2;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
pcapfix:
<details>


warning: unclear-gpl
`gpl3` is a deprecated license, check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/tools/networking/pcapfix/default.nix:16:5:

```
   |
16 |     license = licenses.gpl3;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
podiff:
<details>


warning: patch-phase
`patchPhase` should not be overridden, use `postPatch` instead.

Near pkgs/tools/text/podiff/default.nix:12:3:

```
   |
12 |   patchPhase = ''
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/patch-phase.md
</details>
</details>

**Review by SuperSandro2000 — APPROVED** [review]
LGTM after the last commit is squashed and the other note is anwsered/resolved.

**Comment by siraben** [comment]
@SuperSandro2000 resolved merge conflict.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 115051` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
18 packages built:
</summary>
<ul>
<li>aespipe</li>
<li>agedu</li>
<li>anthy</li>
<li>boolstuff</li>
<li>cconv</li>
<li>clips</li>
<li>cpptest</li>
<li>cpputest</li>
<li>dadadodo</li>
<li>direvent</li>
<li>dos2unix</li>
<li>fanficfare</li>
<li>libdsk</li>
<li>loadwatch</li>
<li>mednafen-server</li>
<li>pcapfix</li>
<li>podiff</li>
<li>vimv</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
aespipe:
<details>


warning: unclear-gpl
`gpl2` is a deprecated license, check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/tools/security/aespipe/default.nix:25:5:

```
   |
25 |     license = licenses.gpl2;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
</details>

**Pull request merged**
